packages/ckeditor5-engine/src/conversion/downcastdispatcher.js

```diff
@@ -118,206 +118,88 @@ export default class DowncastDispatcher {
-	 * @param {module:engine/view/downcastwriter~DowncastWriter} writer View writer that should be used to modify view document.
-	 */
-	convertAttribute( range, key, oldValue, newValue, writer ) {
-		this.conversionApi.writer = writer;
-
-		// Create a list with attributes to consume.
-		this.conversionApi.consumable = this._createConsumableForRange( range, `attribute:${ key }` );
-
-		// Create a separate attribute event for each node in the range.
-		for ( const value of range ) {
-			const item = value.item;
-			const itemRange = Range._createFromPositionAndShift( value.previousPosition, value.length );
-			const data = {
-				item,
-				range: itemRange,
-				attributeKey: key,
-				attributeOldValue: oldValue,
-				attributeNewValue: newValue
-			};
-
-			this._testAndFire( `attribute:${ key }`, data );
-		}
-
-		this._clearConversionApi();
-	}
-
-	/**
-	 * Starts the reconversion of an element. It will:
-	 *
-	 * * Fire an {@link #event:insert `insert` event} for the element to reconvert.
-	 * * Fire an {@link #event:attribute `attribute` event} for element attributes.
-	 *
-	 * This will not reconvert children of the element if they have existing (already converted) views. For newly inserted child elements
-	 * it will behave the same as {@link #convertInsert}.
-	 *
-	 * Element reconversion is defined by the `triggerBy` configuration for the
-	 * {@link module:engine/conversion/downcasthelpers~DowncastHelpers#elementToElement `elementToElement()`} conversion helper.
+	 * Starts a conversion of a model range and the provided markers.
 	 *
 	 * @fires insert
 	 * @fires attribute
-	 * @param {module:engine/model/element~Element} element The element to be reconverted.
+	 * @fires addMarker
+	 * @param {module:engine/model/range~Range} range The inserted range.
+	 * @param {Map<String,module:engine/model/range~Range>} markers The map of markers that should be down-casted.
 	 * @param {module:engine/view/downcastwriter~DowncastWriter} writer The view writer that should be used to modify the view document.
+	 * @param {Object} [options] Optional options object passed to `convertionApi.options`.
 	 */
-	reconvertElement( element, writer ) {
-		const elementRange = Range._createOn( element );
-
-		this.conversionApi.writer = writer;
-
-		// Create a list of things that can be consumed, consisting of nodes and their attributes.
-		this.conversionApi.consumable = this._createInsertConsumable( elementRange );
-
-		const mapper = this.conversionApi.mapper;
-		const currentView = mapper.toViewElement( element );
-
-		// Remove the old view but do not remove mapper mappings - those will be used to revive existing elements.
-		writer.remove( currentView );
-
-		// Convert the element - without converting children.
-		this._convertInsertWithAttributes( {
-			item: element,
-			range: elementRange
-		} );
-
-		const convertedViewElement = mapper.toViewElement( element );
+	convert( range, markers, writer, options = {} ) {
```

Comment:
Do we need to pass markers here or can they be taken from `range` (`range.root` to be precise). I am not sure if `RootElement` has a reference to `Model` instance. Wouldn't it be nice if you didn't have to evaluate those markers on your own (like in data processor)?

Reply:
The problem there is for example with `Title` plugin that for `getBody()` finds only intersecting parts of markers.

Reply:
And in `DataController` we pass there only `_getMarkersRelativeToElement`.

Reply:
Couldn't we evaluate it in the dispatcher? Relative to range in this case, I guess, or let passing range or element to the dispatcher?